projects/backend/capi/articles.ts

```diff
@@ -283,25 +288,87 @@ const parseArticleResult = async (
     }
 }
 
-const capiApiKey = process.env.CAPI_KEY
+type CAPIEndpoint = 'preview' | 'printsent' | 'live'
+
+/**
+ * To access the capi Preview endpoint we need to sign requests (as it is a private api gateway endpoint)
+ * This function generates the necessary headers.
+ * @param endpoint
+ */
+export function sign(endpoint: string) {
+    const url = new URL(endpoint)
+    const credentials = process.env.capiAccessArn
+        ? new ChainableTemporaryCredentials({
+              params: {
+                  RoleArn: process.env.capiAccessArn as string,
+                  RoleSessionName: 'capi-assume-role-access',
+              },
+          })
+        : new SharedIniFileCredentials({ profile: 'capi' })
+
+    const opts = {
+        region: 'eu-west-1',
+        service: 'execute-api',
+        host: url.hostname,
+        path: url.pathname + url.search,
+    }
+
+    const { headers } = new RequestSigner(opts, credentials).sign()
+
+    return headers
+}
+
+const getEndpoint = (capi: CAPIEndpoint, paths: string[]): string => {
+    const queryString = `?ids=${paths.join(',')}&api-key=${
+        process.env.CAPI_KEY
+    }&format=thrift&show-elements=all&show-atoms=all&show-rights=all&show-fields=all&show-tags=all&show-blocks=all&show-references=all&page-size=100`
+
+    switch (capi) {
+        case 'printsent':
+            return `${process.env.psurl}/search${queryString}`
+        case 'live':
+            return `https://content.guardianapis.com/search${queryString}`
+        case 'preview':
+            return `${process.env.capiPreviewUrl}/search${queryString}`
+        default:
+            return ''
+    }
+}
+
+const getPreviewHeaders = (endpoint: string) => {
+    return {
+        Accept: 'application/json',
+        ...sign(endpoint),
+    }
+}
 
-const printsent = (paths: string[]) =>
-    `${process.env.psurl}?ids=${paths.join(
-        ',',
-    )}&format=thrift&api-key=${capiApiKey}&show-elements=all&show-atoms=all&show-rights=all&show-fields=all&show-tags=all&show-blocks=all&show-references=all&format=thrift&page-size=100`
+const isScheduledSoon = (dateiso8601: string): boolean => {
```

Comment:
Rename the method to reflect what it is doing? maybe something like `scheduledWithinAMonth`?